installers/custom_ipxe/main.go

```diff
@@ -3,22 +3,71 @@ package custom_ipxe
 import (
 	"strings"
 
+	"github.com/packethost/pkg/log"
+	"github.com/tinkerbell/boots/installers"
 	"github.com/tinkerbell/boots/ipxe"
 	"github.com/tinkerbell/boots/job"
+	"github.com/tinkerbell/boots/packet"
 )
 
 func init() {
-	job.RegisterSlug("custom_ipxe", bootScript)
+	job.RegisterInstaller("ipxe", ipxeScript)
+	job.RegisterSlug("custom_ipxe", ipxeScript)
 }
 
-func bootScript(j job.Job, s *ipxe.Script) {
+func ipxeScript(j job.Job, s *ipxe.Script) {
+	logger := installers.Logger("custom_ipxe")
+	if j.InstanceID() != "" {
+		logger = logger.With("instance.id", j.InstanceID())
+	}
+
+	var cfg *packet.InstallerData
+	var err error
+
+	if j.OperatingSystem().Installer == "ipxe" {
+		cfg = j.OperatingSystem().InstallerData
+		if cfg == nil {
+			s.Echo("Installer data not provided")
+			s.Shell()
+			logger.Error(err, "empty installer data")
+
+			return
+		}
+	} else {
+		if strings.HasPrefix(j.UserData(), "#!ipxe") {
+			cfg = &packet.InstallerData{Script: j.UserData()}
+		} else {
+			cfg = &packet.InstallerData{Chain: j.IPXEScriptURL()}
+		}
+	}
+
+	IpxeScriptFromConfig(logger, cfg, j, s)
+}
+
+func IpxeScriptFromConfig(logger log.Logger, cfg *packet.InstallerData, j job.Job, s *ipxe.Script) {
```

Comment:
should this function be unexported?

Reply:
good point, this doesn't need to be anymore. Will fix